fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/api/DelinquencyApiResource.java

```diff
@@ -0,0 +1,217 @@
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Retrieve a specific Delinquency Range based on the Id", description = "")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = DelinquencyApiResourceSwagger.GetDelinquencyRangesResponse.class))) })
+    public String getDelinquencyRange(
+            @PathParam("delinquencyRangeId") @Parameter(description = "delinquencyRangeId") final Long delinquencyRangeId,
+            @Context final UriInfo uriInfo) {
+        securityContext.authenticatedUser().validateHasReadPermission("DELINQUENCY_BUCKET");
+        final DelinquencyRangeData delinquencyRangeData = this.readPlatformService.retrieveDelinquencyRange(delinquencyRangeId);
+        ApiRequestJsonSerializationSettings settings = parameterHelper.process(uriInfo.getQueryParameters());
+        return this.jsonSerializerRange.serialize(settings, delinquencyRangeData);
+    }
+
+    @POST
+    @Path("ranges")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Create Delinquency Range", description = "")
+    @RequestBody(required = true, content = @Content(schema = @Schema(implementation = DelinquencyApiResourceSwagger.PostDelinquencyRangeRequest.class)))
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = DelinquencyApiResourceSwagger.PostDelinquencyRangeResponse.class))) })
+    public String createDelinquencyRange(final String jsonRequestBody, @Context UriInfo uriInfo) {
+        securityContext.authenticatedUser().validateHasCreatePermission("DELINQUENCY_BUCKET");
+        final CommandWrapper commandRequest = new CommandWrapperBuilder().createDelinquencyRange().withJson(jsonRequestBody).build();
+
+        CommandProcessingResult result = commandWritePlatformService.logCommandSource(commandRequest);
+        return jsonSerializerRange.serialize(result);
+    }
+
+    @PUT
+    @Path("ranges/{delinquencyRangeId}")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Update Delinquency Range based on the Id", description = "")
+    @RequestBody(required = true, content = @Content(schema = @Schema(implementation = DelinquencyApiResourceSwagger.PostDelinquencyRangeRequest.class)))
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = DelinquencyApiResourceSwagger.PutDelinquencyRangeResponse.class))) })
+    public String updateDelinquencyRange(
+            @PathParam("delinquencyRangeId") @Parameter(description = "delinquencyRangeId") final Long delinquencyRangeId,
+            final String jsonRequestBody, @Context UriInfo uriInfo) {
+        securityContext.authenticatedUser().validateHasCreatePermission("DELINQUENCY_BUCKET");
```

Comment:
@josehernandezfintecheandomx Wrong permission got checked. It should use the "validateHasUpdatePermission"

Reply:
Done!

Reply:
@josehernandezfintecheandomx it looks the same to me :/
`validateHasCreatePermission`